lib/http/client.cpp

```diff
@@ -25,15 +25,22 @@ Client::Client(const std::string &server, uint16_t port)
 void
 Client::set_server(const std::string &server)
 {
+        std::string server_name = server;
+        // Remove https prefix, if it exists
+        if (boost::algorithm::starts_with(server_name, "https://"))
```

Comment:
I wouldn't. That would mean, we are sending plain text passwords over the network, etc. Also this is currently just a convenience "helper", because synapse asks you to use this https address, when you login. I'm open to change this, when any user has an issue with this, but for now I would require https. We also don't have any servers to test this, I think.